apps/bugs_bunny/lib/bugs_bunny/worker/rabbit_connection.ex

```diff
@@ -135,9 +137,36 @@ defmodule BugsBunny.Worker.RabbitConnection do
   end
 
   @impl true
-  def handle_info(:connect, %{config: config, config: config} = state) do
-    get_client(config).open_connection(config)
-    |> handle_rabbit_connect(state)
+  def handle_info(:connect, %{adapter: adapter, config: config} = state) do
+    # TODO: add exponential backoff for reconnects
```

Comment:
i prefer moving this to its own method but i think is a matter of taste 👍

Reply:
I just moved the comment when I inlined the code :)

Reply:
ah, I just re-read your comment; I think you are referring to the code below your comment.

The reason I like it this way is that I prefer the return expression to be in the same block as the `handle_*` function definition. So often I call out to a `case` and pattern match on the result so I can return the `{:noreply, …}`. It is true you can have that returned from a helper function, but that will make it harder for the reader to know what is going on as the reader will have to follow the code path to figure out what will happen.

Does that make sense? Perhaps I should write a blogpost about this :)